### ISSUE TITLE
Why does traffic analysis output differ from the video in readme file?

### ISSUE THREAD

**Issue by INF800** [roboflow/supervision#1594]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

I tried running the same code in readme to generate video outputs but accuracy is not upto mark.

Here are videos with different thresholds: 

https://drive.google.com/drive/folders/1TFcEJcSvVSQXaMEYQTnhw2s-QNXP-LOz?usp=sharing

**Comment by INF800** [comment]
FYI, I used roboflow API instead of running it locally.

cc: @onuralpszr

**Comment by INF800** [comment]
I see that default model id is `vehicle-count-in-drone-video/6` was it supposed to be `8` instead of `6`

cc: @onuralpszr

**Comment by INF800** [comment]
8 is giving resource error

**Comment by onuralpszr** [comment]
> I see that default model id is `vehicle-count-in-drone-video/6` was it supposed to be `8` instead of `6`
> 
> cc: @onuralpszr

That is dataset not model so default 6 is correct

**Comment by onuralpszr** [comment]
> 8 is giving resource error
 
 v8 as model not exist so it is correct

**Comment by onuralpszr** [comment]
> FYI, I used roboflow API instead of running it locally.
> 
> cc: @onuralpszr

Out of curiosity what is your inference version ?

**Comment by INF800** [comment]
It's the same as in requirements.txt file

<img width="989" alt="Screenshot 2024-10-14 at 7 14 17 AM" src="https://github.com/user-attachments/assets/2d6be7e9-9c44-4ba3-bc58-2992854d165d">

**Comment by INF800** [comment]
I got this warning message before running the script:

```
[10/14/24 07:10:56] WARNING  Your inference package version 0.9.17 is out of date! Please upgrade to version 0.23.0 of inference for    __init__.py:35
                             the latest features and bug fixes by running `pip install --upgrade inference`.
```

but ignored it anyway because that's what is being used in requirements.txt

cc: @onuralpszr

**Comment by INF800** [comment]
Just now I trained a new model using my custom data (just 2 4k images) and its results are wayyy better that current one

**Comment by INF800** [comment]
> Just now I trained a new model using my custom data (just 2 4k images) and its results are wayyy better that current one

here are the results: https://drive.google.com/drive/folders/1bzHpSBSRi3nbFa7i-BdMZjTRbpqnp3nc?usp=sharing

ccL @onuralpszr

**Comment by onuralpszr** [comment]
@PawelPeczek-Roboflow can you help me out on this one please I was able to generate the same problematic video and  I tried version in written req.txt and I tried latest version of inference to test and my thought was current code is not using conf and iou values then I used latest version with parameters I checked in documentation and still same results. Either model is roboflow problematic or inference code doing something weird 

Latest version of inference changes I made in example code (code snippet not entire code) 
```python
from inference import get_model
...
self.model = get_model(model_id=model_id, api_key=roboflow_api_key)
...
results = self.model.infer(
    image=frame, confidence=self.conf_threshold, iou_threshold=self.iou_threshold
)[0]
```

cc @LinasKo 
cc @SkalskiP  (since you wrote initially you might be have better understanding of model side)

**Comment by onuralpszr** [comment]
> I got this warning message before running the script:
> 
> ```
> [10/14/24 07:10:56] WARNING  Your inference package version 0.9.17 is out of date! Please upgrade to version 0.23.0 of inference for    __init__.py:35
>                              the latest features and bug fixes by running `pip install --upgrade inference`.
> ```
> 
> but ignored it anyway because that's what is being used in requirements.txt
> 
> cc: @onuralpszr

I tested on both cases yes and I did the same thing too

**Comment by onuralpszr** [comment]
> > Just now I trained a new model using my custom data (just 2 4k images) and its results are wayyy better that current one
> 
> here are the results: https://drive.google.com/drive/folders/1bzHpSBSRi3nbFa7i-BdMZjTRbpqnp3nc?usp=sharing
> 
> ccL @onuralpszr

You used inference local model run right ?

**Comment by INF800** [comment]
> > > Just now I trained a new model using my custom data (just 2 4k images) and its results are wayyy better that current one
> > 
> > 
> > here are the results: https://drive.google.com/drive/folders/1bzHpSBSRi3nbFa7i-BdMZjTRbpqnp3nc?usp=sharing
> > ccL @onuralpszr
> 
> You used inference local model run right ?

Nope, used Roboflow API with the same inference code as is with same version dependencies. The only change that I made is replace the model is with my newly trained model ID.

Based on this I think everything working just fine at code level.

**Comment by onuralpszr** [comment]
> > > > Just now I trained a new model using my custom data (just 2 4k images) and its results are wayyy better that current one
> > > 
> > > 
> > > here are the results: https://drive.google.com/drive/folders/1bzHpSBSRi3nbFa7i-BdMZjTRbpqnp3nc?usp=sharing
> > > ccL @onuralpszr
> > 
> > You used inference local model run right ?
> 
> Nope, used Roboflow API with the same inference code as is with same version dependencies. The only change that I made is - replacing the model id with my newly trained model ID.
> 
> Based on this I think everything working just fine at code level.
> 

For code level yes It should be fine that I am sure I am converting this to Q&A discussion and continue from there. 

Also you could you share your ID if it is public please

**Comment by INF800** [comment]
Yes the one that I used for this `cusom-data/2` (with the typo).

cc: @onuralpszr 

I'm using the model for running some inference jobs right now (15mins more remaining).

**Comment by PawelPeczek-Roboflow** [comment]
@onuralpszr  do not really know the context, could you elaborate a little bit more?

**Comment by LinasKo** [comment]
Hi @INF800 👋 

Based on the video, it's likely that InferenceSlicer was used to help detect small objects. Have you tried using it?
([guide](https://supervision.roboflow.com/latest/how_to/detect_small_objects/), [docs](https://supervision.roboflow.com/latest/detection/tools/inference_slicer/))

Next, it would help if you could share some code you used to run the model. Specifically, I'm curious what code you ran that produced poor results initially.

**Comment by onuralpszr** [comment]
> @onuralpszr do not really know the context, could you elaborate a little bit more?

In basic context, inference example in here https://github.com/roboflow/supervision/tree/develop/examples/traffic_analysis doesn't show output of that example video in readme.

**Comment by PawelPeczek-Roboflow** [comment]
ok, taken a look - first of all, the model trained for `ultralytics` is a different class of model than the pointed one from Roboflow - not sure if the results were comparable in first place

**Comment by LinasKo** [comment]
Hey @PawelPeczek-Roboflow, what do you mean by "a different class of model"? What kind of difference are we looking at? Model family, size, or a different dataset?

**Comment by PawelPeczek-Roboflow** [comment]
not fully sure about details, but the weights size differ:
* one order of magnitude in favour of ultralytics model (it also runs 5x slower)
* additionally model from universe was trained on 100 images - we shall not expect it to be robust

**Comment by PawelPeczek-Roboflow** [comment]
also the Roboflow model was not that great in the metrics of original dataset
![image](https://github.com/user-attachments/assets/f5fa5f01-fa1a-4973-b62e-1380630b823c)

**Comment by LinasKo** [comment]
Curious.

@INF800, all data points to the video as something used as an example only. If you wish to get something of a similar performance, I suggest using one of yolov8 models from [inference](https://inference.roboflow.com/quickstart/aliases/). If the objects are this small, I also suggest Inference Slicer ([guide](https://supervision.roboflow.com/latest/how_to/detect_small_objects/), [docs](https://supervision.roboflow.com/latest/detection/tools/inference_slicer/)).

I'll mark this as potentially misleading and investigate further later on. Thank you for reporting it.
